polar-core/tests/integration_tests.rs

```diff
@@ -988,45 +985,40 @@ fn test_anonymous_vars() {
 
 #[test]
 fn test_singleton_vars() {
-    let (mut read, write) = pipe();
-    spawn(move || {
-        let mut polar = Polar::new(Some(Box::new(write)));
-        polar.register_constant(sym!("X"), term!(true));
-        polar.register_constant(sym!("Y"), term!(true));
-        polar.load("f(x:X,y:Y,z:Z) if z = z;").unwrap();
-    });
-
-    let mut out = String::new();
-    read.read_to_string(&mut out).unwrap();
-    assert_eq!(
-        &out,
-        indoc!(
-            r#"Singleton variable x is unused or undefined, see <https://docs.oso.dev/using/polar-syntax.html#variables>
-               001: f(x:X,y:Y,z:Z) if z = z;
-                      ^
-               Singleton variable y is unused or undefined, see <https://docs.oso.dev/using/polar-syntax.html#variables>
-               001: f(x:X,y:Y,z:Z) if z = z;
-                          ^
-               Unknown specializer Z
-               001: f(x:X,y:Y,z:Z) if z = z;
-                                ^
-               "#
-        )
-    );
+    let messages = MessageQueue::new();
+    let mut polar = Polar::new(Some(messages.clone()));
+    polar.register_constant(sym!("X"), term!(true));
+    polar.register_constant(sym!("Y"), term!(true));
+    polar.load("f(x:X,y:Y,z:Z) if z = z;").unwrap();
+    let output = messages.next().unwrap();
+    assert!(matches!(&output.kind, MessageKind::Warning));
+    // @TODO: How does this work?
```

Comment:
What's up with this commented-out block?

Reply:
fixed